### ISSUE TITLE
About the .mat files under metric_init

### ISSUE THREAD

**Issue by chungweihang** [gaohuang/S-WMD#3]
Hi @gaohuang @mkusner,

Thanks for the good work.  I am trying to run it on some news dataset and wondering how to produce the files under `metric_init`.   Could you say about those files and their fields?

Thanks,
Chung-Wei

**Comment by gaohuang** [comment]
Hi, the preprocessed dataset need to contain the following fields:

X is a cell array of all documents, each represented by a dxm matrix where d is the dimensionality of the word embedding and m is the number of unique words in the document
Y is an array of labels
BOW_X is a cell array of word counts for each document
indices is a cell array of global unique IDs for words in a document
TR is a matrix whose ith row is the ith training split of document indices
TE is a matrix whose ith row is the ith testing split of document indices

**Comment by chungweihang** [comment]
What about these files? https://github.com/gaohuang/S-WMD/tree/master/metric_init

**Comment by gaohuang** [comment]
Those files contain the initial projection matrix A_init learned with NCA, which is implemented by this toolbox https://lvdmaaten.github.io/drtoolbox/ and many others).

**Comment by chungweihang** [comment]
Hi, one more question: can you say more about the `makesplits_seed` function in [`scale_metric.m`](https://github.com/gaohuang/S-WMD/blob/master/scale_metric.m)?  Thanks

**Comment by gaohuang** [comment]
The makesplit function is used for splitting the dataset into a training set and a test/validation set.

The scale_metric function is used to scale the projection matrix A_init by a scalar to make the sinkhorn algorithm numerically stable. This is just a heurisitc. You can skip this step, but you may need more effort to search for a proper value for lambda.

**Comment by mkusner** [comment]
Thanks a lot Gao!

The last thing I'll add is that to learn A_init, you start by taking, for
each document, all of the word vectors for the words that appear in that
document. You then multiply each word vector by its normalized BOW weight
in that document. You then sum all of these word vectors. This is
essentially a weighted average vector for each document (and it is the
representation used by the WCD distance in the original word mover's
distance paper: http://mkusner.github.io/publications/WMD.pdf). Then you
simply train NCA as Gao described on these weighted average vectors, as you
would on any dataset. This will give you A_init.

**Comment by chungweihang** [comment]
Thank you both.  I am asking this because I was able to generate A_init with NCA, but when I try to run `scale_metric`, it can't find `makesplits_seed` function in the repository.  I might try to just skip `scale_metric` as Gao suggested.

**Comment by mkusner** [comment]
Oh I see!

Right you can certainly skip scale_metric. Or if you want to run it you can
replace line 22 in scale_metric.m with:

[idx_tr, idx_val] = makesplits(ytr, 0.8, 1, 1, 1);

The makesplits_seed function lets you specify a random seed so that the
splits are always the same, but in this case I don't think you need that so
the makesplits function (in the /functions directory) should be just fine.
Alternatively you can add an additional initial argument to makesplits
where you can pass in a seed and then just set the random seed at the start
of the function with that seed and you have makesplits_seed  :)

**Comment by gaohuang** [comment]
Thank you Matt! The explanation is very clear.

@chungweihang Thanks for pointing us to this problem. Once you get the code running, could you please create a pull request?
